modules/torch_utils.py

```diff
@@ -21,6 +21,6 @@ def get_param(model) -> torch.nn.Parameter:
 def float64(t: torch.Tensor):
     """return torch.float64 if device is not mps or xpu, else return torch.float32"""
     match t.device.type:
-        case 'mps', 'xpu':
+        case ('mps' | 'xpu'):
```

Comment:
oh yeah thanks that was my blunder

the parentheses is redundant
just this should work
```py
case 'mps' | 'xpu':
```

Reply:
Done!